src/Data/Array/Mutable/Linear.hs

```diff
@@ -111,8 +112,8 @@ fromList list (f :: Array a #-> Unrestricted b) =
 -- # Mutations and Reads
 -------------------------------------------------------------------------------
 
-size :: Array a #-> (Array a, Int)
-size (Array size' arr) =  (Array size' arr, size')
+size :: Array a #-> (Array a, Unrestricted Int)
+size (Array size' arr) =  (Array size' arr, Unrestricted size')
```

Comment:
Careful to rebase if the `Unrestricted -> Ur` change is merged before (and vice versa).